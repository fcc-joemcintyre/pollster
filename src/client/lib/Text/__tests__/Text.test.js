/* eslint-env jest */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { Text } from '..';

const theme = {
  fonts: {
    normal1: `
      font-family: 'Open Sans', sans-serif;
      font-weight: 400;
    `,
    bold1: `
      font-family: 'Open Sans', sans-serif;
      font-weight: 600;
    `,
    normal2: `
      font-family: 'Lato', sans-serif;
      font-weight: 400;
    `,
    bold2: `
      font-family: 'Lato', sans-serif;
      font-weight: 700;
    `,
  },
};

test ('Text default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Text>Some text</Text>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Text font 1 normal', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Text font='normal1'>Some text</Text>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Text font 1 bold', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Text font='bold1'>Some text</Text>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Text font 2 normal', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Text font='normal2'>Some text</Text>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Text font 2 bold', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <Text font='bold2'>Some text</Text>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
