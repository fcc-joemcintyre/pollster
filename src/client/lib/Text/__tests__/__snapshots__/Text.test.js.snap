// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text default render 1`] = `
.c0 {
  font-family: sans-serif;
  font-weight: normal;
  font-size: 16px;
  display: block;
}

<div
  className="c0"
>
  Some text
</div>
`;

exports[`Text font 1 bold 1`] = `
.c0 {
  font-family: 'Open Sans',sans-serif;
  font-weight: 600;
  font-size: 16px;
  display: block;
}

<div
  className="c0"
>
  Some text
</div>
`;

exports[`Text font 1 normal 1`] = `
.c0 {
  font-family: 'Open Sans',sans-serif;
  font-weight: 400;
  font-size: 16px;
  display: block;
}

<div
  className="c0"
>
  Some text
</div>
`;

exports[`Text font 2 bold 1`] = `
.c0 {
  font-family: 'Lato',sans-serif;
  font-weight: 700;
  font-size: 16px;
  display: block;
}

<div
  className="c0"
>
  Some text
</div>
`;

exports[`Text font 2 normal 1`] = `
.c0 {
  font-family: 'Lato',sans-serif;
  font-weight: 400;
  font-size: 16px;
  display: block;
}

<div
  className="c0"
>
  Some text
</div>
`;
